pointcloud/basic_octree.go

```diff
@@ -198,22 +197,21 @@ func (octree *BasicOctree) ToProtobuf() *commonpb.Geometry {
 	return nil
 }
 
-// CollidesWithGeometry will return whether a given geometry is in collision with a given point.
-// A point is in collision if its stored probability is >= confidenceThreshold and if it is at most buffer distance away.
-func (octree *BasicOctree) CollidesWithGeometry(
-	geom spatialmath.Geometry,
-	confidenceThreshold int,
-	buffer,
-	collisionBufferMM float64,
-) (bool, error) {
+// CollidesWith checks if the given octree collides with the given geometry and returns true if it does.
+// A point is in collision if its stored probability is >= confidenceThreshold and if it is at most collisionBufferMM distance away.
+func (octree *BasicOctree) CollidesWith(geom spatialmath.Geometry, collisionBufferMM float64) (bool, error) {
 	if octree.MaxVal() < confidenceThreshold {
```

Comment:
I worry about this.

It's easy to naively construct an octree from points, where points have a value of either zero or MinInt64, both of which will cause difficult to diagnose problems.

I wonder if we should remove this check entirely; we cannot know what the user's criteria for "confidence" is, and it might be more appropriate to simply expect the caller to pre-filter the octree to remove points below their confidence appetite.

Reply:
Would it make you feel better if there was a way to configure this somehow? Or would you rather this just get removed entirely

Reply:
I think I prefer to remove entirely.

There are a lot of things a user might want to filter a pointcloud on; confidence value, color, position, etc.

Seeing as we do not have any hard requirements whatsoever on what "confidence" means for a pointcloud, e.g. enforcing that it must be between 0-100, this is really just a secret magic number that breaks things sometimes.

Reply:
My only concern is for how this will impact SLAM based motion and I have no idea what the answer to that is

Reply:
Actually I just checked with John this will absolutely break SLAM stuff.  I'm going to make it configurable and if left unset will not apply this check